client/src/main/java/org/apache/celeborn/client/write/PushState.java

```diff
@@ -88,4 +97,22 @@ public boolean addBatchData(String addressPair, PartitionLocation loc, int batch
   public DataBatches takeDataBatches(String addressPair) {
     return batchesMap.remove(addressPair);
   }
+
+  public void addFlightBatches(int batchId, PartitionLocation loc) {
+    String addressPair = loc.hostAndPushPort();
+    Set<Integer> batchIdSetPerPair =
+        batchIdPerAddressPair.computeIfAbsent(addressPair, id -> new HashSet<>());
+    batchIdSetPerPair.add(batchId);
+  }
+
+  public void removeFlightBatches(int batchId, PartitionLocation loc) {
+    String hostAndPushPort = loc.hostAndPushPort();
+    Set<Integer> batchIdSetPerPair = batchIdPerAddressPair.get(hostAndPushPort);
+    if (Objects.nonNull(batchIdSetPerPair)) {
```

Comment:
Which case loc can be null?

Reply:
Technically, the loc will not be null, it's just my coding habits.